extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -167,8 +183,20 @@ public SecurityIdentity apply(Throwable throwable) {
             }
         }
 
-        challenge = new ChallengeData(HttpResponseStatus.FOUND.code(), HttpHeaders.LOCATION,
-                configContext.auth.authorizeURL(params));
+        if (shouldAutoRedirect(configContext, context)) {
+            challenge = new ChallengeData(HttpResponseStatus.FOUND.code(), HttpHeaders.LOCATION,
+                    configContext.auth.authorizeURL(params));
+        } else {
+            // If the client (usually an SPA) wants to handle the redirect manually, then
+            // return status code 499 and in the www-authenticate header put the redirect url
+            // www-authenticate header content; No standard content matched the exact use case
+            // so the OAuth redirect-url= is an arbitrary string that can be tested
+            // for on the client side
+            String headerContent = "OIDC redirect-url=" + configContext.auth.authorizeURL(params);
```

Comment:
@boosey, so if we want the users to do `location.reload` on seeing `499` then we don't really need `OIDC redirect-url` ?

Reply:
I was just folllowing the guidance I found on the purpose and usage of WWW-Authenticate.

Reply:
@boosey Right, but this `499` is a workaround as opposed to a genuine `401` which is when `WWW-Authenticate` is required, i.e it is the authorization code flow. Hmm. Well, since we are transporting this `499` with `ChallengeData` then may be it is better to at least provide `OIDC` as the supported authentication method...But I propose to drop ` redirect-url` (I did propose to return some hint myself - simply because I was not quite getting initially what was happening in the PR :-)  ), we don't really want the user code starting doing something manual around this redirect url, etc...

Reply:
Ok. I will remove the redirect-url.